test/model/recent_senders_test.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/model/recent_senders.dart';
+import '../example_data.dart' as eg;
+
+void main() {
+  group('MessageIdTracker', () {
+    late MessageIdTracker idTracker;
+
+    void prepare() {
+      idTracker = MessageIdTracker();
+    }
+
+    test('starts with no ids', () {
+      prepare();
+      check(idTracker.debugIds).isEmpty();
+    });
+
+    test('calling add(id) adds the same id to the tracker just one time', () {
+      prepare();
+      check(idTracker.debugIds).isEmpty();
+      idTracker.add(1);
+      idTracker.add(1);
+      check(idTracker.debugIds.singleOrNull).equals(1);
+    });
+
+    test('ids are sorted ascendingly, with maxId pointing to the last element', () {
+      prepare();
+      check(idTracker.debugIds).isEmpty();
+      idTracker.add(1);
+      idTracker.add(9);
+      idTracker.add(0);
+      idTracker.add(5);
+      check(idTracker.debugIds).deepEquals([0, 1, 5, 9]);
+      check(idTracker.maxId).equals(idTracker.debugIds.last);
+    });
+  });
+
+  group('RecentSenders', () {
```

Comment:
This has no tests for the methods the app relies on for reading data from this class :slightly_smiling_face:. They won't be complicated, but there should be tests for `latestMessageIdOfSenderInStream` and its topic counterpart.

In particular those tests should exercise when there are no such messages — it should detect the difference between returning null for that case, and returning -1 as the interface was in a previous revision of this PR.

Also should exercise when the stream is present but not the topic, and when the stream is absent.